invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -120,7 +134,13 @@ def view(**kwargs):
                 except NoResultFound:
                     record = service().read(**read_kwargs)
             else:
-                record = service().read(**read_kwargs)
+                try:
+                    read_kwargs["front_end_call"] = True
```

Comment:
what is the need of this `front_end_call`?

Reply:
So essentially
when going through /api/
the only function in `invenio-records-resources `which is called is search (and this works fine)
the issue is that when the call is done through `app-rdm`
it first calls `records.search` on `invenio-records-resources`, then calls `files.search` 

if we throw an error with the correct pid in `records.search`, then the call to /api wil fail (but `app-rdm` will not as the error is caught)
if we don't throw an error, then `app-rdm` will try again to access a record using the wrong id (ie not of the most recent parent) and this crashes in` files.search `(one solution would be to also resolve in `files.search` the parent pid in order to find the most recent), the issue is that this could also overlap on other files

finaly going back to the original question
if we add a flag to the function, we are able to distinguish wether it is a backend (/api) or frontend call (app-rdm)